# Structural bioinformatics practical

**Part II BBS Bioinformatics** - 16:00-18:00, 5 Mar 2018

Instructions for the practical are below. [Answers](Answers.md) are available for reference.

Just in case, I've made available all the [files](files) that you are otherwise supposed to download throughout the practical from their original sources.

---
1. **Finding the right structures - UniProt and PDB.**

We are interested in analysing the structure of a protein called arginine-tRNA ligase (synthetase) from the yeast _Saccharomyces cerevisiae_.

a) Go to [UniProt](http://www.uniprot.org/) and find the entry corresponding to this protein using the search bar. Take a look around at the available information. Feel free to click through to various databases to familiarise yourself with what information UniProt aggregates.

b) Download the `.fasta` file containing the sequence of our protein. We'll need it later.

c) Note down the residue numbers interacting with ligands. We'll visualise them later.



2. **Interesting databases.**

Go back to the UniProt entry for our protein.

a) Click through to the InterPro database entry (`CTRL-F` for "View protein in InterPro").
- What domains and motifs does our protein contain, and are they well supported?
- Does it match what you see in the structure?
- What InterPro family is it part of? How many members does it have and is our protein similar to the most common domain architecture in this family or is it part of a smaller subgroup?

b) From the "Structures" tab, follow the links to SCOP and CATH database entries.

c)
